src/material/core/tokens/_token-utils.scss

```diff
@@ -406,3 +421,71 @@ $_component-prefix: null;
   }
   @return $result;
 }
+
+/// Returns a theme config with the given tokens overridden.
+/// @param {Map} $theme The material theme for an application.
+/// @param {Map} $overrides The token values to override in the theme.
+@function extend-theme($theme, $extend-namespaces, $overrides: ()) {
+  $internals: _mat-theming-internals-do-not-access;
+  $systems: (color-tokens, typography-tokens, density-tokens, base-tokens);
+  $variants: (primary, secondary, tertiary, error, surface);
+
+  @if (inspection.get-theme-version($theme) < 1) {
+    @error #{'The `extend-theme` functions are only supported for M3 themes'};
+  }
+
+  // Determine which system and namespace each shorthand token belongs to.
+  $seen-tokens: ();
+  @each $namespace in $extend-namespaces {
+    @each $system in $systems {
+      $tokens: map.get($theme, $internals, $system, $namespace);
+      @if $tokens == null {
+        @error #{'Error extending theme: Theme does not have tokes for namespace `('}#{$namespace}#{
+          ')`'};
+      }
+      @each $name, $value in $tokens {
+        @if map.has-key($seen-tokens, $name) {
+          @error #{'Error extending theme: Ambiguous token name `'}#{$name}#{
```

Comment:
They shouldn't, because it doesn't depend on the overrides the user passes in. It's based on the namespaces the function is working with, which should be hardcoded for all of the APIs we expose. This error is more to help us identify issues if we e.g. add a new token whose name conflicts with a different namespace handled by the same function.